TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -750,6 +750,21 @@ public virtual void StartExecStatement(ExecStatement execStmt)
 
         public virtual void EndExecStatement()
         {
+            //procedures containing EXEC statements should be generated as nested
```

Comment:
I saw this with @mayanje following the bug reported by @sarterbe.
The event EndExecStatement (therefore StartExecStatement) is only fired when the Exec statement is located in the procedure division and at the top of the instructions, while OnExecStatement is fired when the Exec statement is located in the procedure division and anywhere else but at the top of the instructions.
This behaviour might be a bug we need to fix. The two SetFlag are a quick fix to generate correctly the lines we need.

Reply:
The search for the FunctionDeclaration node is not to make sure we are in a FunctionDeclaration, but rather to set the flag on this declaration that contains an Exec Statement. The test on _InsideProcedure avoids to look for a parent of type FunctionDeclaration if the Exec statement is not into a procedure.
But maybe we could store the FunctionDeclaration node instead of "if we are inside a procedure division" ?

Reply:
Ok, so for the EndExecStatement  and OnExecStatement can you add this explanation directly in the source code?
Otherwise someone could remove the duplication to "optimize" the code.